app/_src/gateway/reference/expressions-language/language-references.md

```diff
@@ -0,0 +1,165 @@
+|----------|------------------------------------------------------------------------------------------------------|------------|---------------|
+| `String` | A string value, always in valid UTF-8.                                                               | ✅          | ✅             |
+| `IpCidr` | Range of IP addresses in CIDR format. Can be either IPv4 and IPv6.                                   | ❌          | ✅             |
+| `IpAddr` | A single IP address. Can be either IPv4 and IPv6.                                                    | ✅          | ✅             |
+| `Int`    | A 64-bit signed integer.                                                                             | ✅          | ✅             |
+| `Regex`  | A regex in [syntax](https://docs.rs/regex/latest/regex/#syntax) specified by the Rust `regex` crate. | ❌          | ✅             |
+
+In addition, Expressions also supports one composite type - `Array`. Array types are written as `Type[]`.
+For example: `String[]`, `Int[]`. Currently arrays can only be present in field values. They are used in
+case one field could contain multiple values. e.g. `http.headers.x` or `http.queries.x`.
+
+### String
+
+Strings are valid UTF-8 sequences. They can be defined with string literal that looks like:
+`"content"`. `\n`, `\r`, `\t`,`\\` as well as `\"` escape sequences are supported:
+
+| Escape sequence | Description               |
+|-----------------|---------------------------|
+| `\n`            | Newline character         |
+| `\r`            | Carriage return character |
+| `\t`            | Horizontal tab character  |
+| `\\`            | The `\` character         |
+| `\"`            | The `"` character         |
+
+In addition, Expressions supports raw string literals which takes form in `r#"content"#`.
+This feature is useful in case you want to write a regex and repeated escape quickly become
+tedious to deal with:
+
+For example, if you want to match `http.path` against `/\d+\-\d+` using the regex `~` operator:
+
+With string literals, the predicate will be written as:
+
+```
+http.path ~ "/\\d+\\-\\d+"
+```
+
+With raw string literals, you can simply write:
+
+```
+http.path ~ r#"/\d+\-\d+"#
+```
+
+### IpCidr
+
+`IpCidr` represents a range of IP addresses in [Classless Inter-Domain Routing (CIDR)](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) format.
+Here are some examples:
+
+IPv4:
+
+```
+net.src.ip in 192.168.1.0/24
+```
+
+IPv6:
+```
+net.src.ip in fd00::/8
+```
+
+> **Note:** Expressions parser rejects any CIDR literal where the host portion contains
+  any non-zero bits. This means that `192.168.0.1/24` won't pass the parser check because
+  the intention of the author is unclear.
+
+### IpAddr
+
+`IpCidr` represents a single IP addresses in [IPv4 Dot-decimal notation](https://en.wikipedia.org/wiki/Dot-decimal_notation),
```

Comment:
```suggestion
`IpAddr` represents a single IP addresses in [IPv4 Dot-decimal notation](https://en.wikipedia.org/wiki/Dot-decimal_notation),
```
typo here?

Reply:
Should be "`IpAddr` represents a single IP address"?